crates/oxc_linter/src/rules/eslint/no_empty_function.rs

```diff
@@ -85,6 +85,8 @@ impl TryFrom<&str> for Allowed {
             "getters" | "getter" => Ok(Self::Getters),
             "setters" | "setter" => Ok(Self::Setters),
             "constructors" | "constructor" => Ok(Self::Constructors),
+            "asyncFunctions" | "async-functions" => Ok(Self::AsyncFunctions),
```

Comment:
We do not need to support `async-*` option.

`private-constructors` and `protected-constructors` are only there, because `typescript-eslint` implemented them before, ESLint did with `protectedConstructors`.

https://typescript-eslint.io/rules/no-empty-function/#allow-protected-constructors
https://eslint.org/docs/latest/rules/no-empty-function#allow-protectedconstructors

https://github.com/eslint/eslint/pull/19551
https://github.com/typescript-eslint/typescript-eslint/pull/1267